multiplexer/appd/run.go

```diff
@@ -203,7 +119,131 @@ func (a *Appd) Stop() error {
 	return nil
 }
 
-// Pid returns the pid of the appd process.
+// Pid returns the process ID of the appd process.
 func (a *Appd) Pid() int {
 	return a.pid
 }
+
+// CreateExecCommand creates an exec.Cmd for the appd binary.
+func (a *Appd) CreateExecCommand(args ...string) *exec.Cmd {
+	cmd := exec.Command(a.path, args...)
+	cmd.Stdin = a.stdin
+	cmd.Stdout = a.stdout
+	cmd.Stderr = a.stderr
+	return cmd
+}
+
+// getPathToBinary returns the path to the celestia-appd binary for the given version.
+func getPathToBinary(version string) (binaryPath string, err error) {
+	baseDirectory := getDirectoryForVersion(version)
+
+	// look for the executable binary in the extracted files
+	err = filepath.Walk(baseDirectory, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !info.IsDir() && info.Mode()&0o111 != 0 {
+			binaryPath = path
+			return filepath.SkipAll // Found it, stop searching
```

Comment:
TIL this is magic :D 

> SkipAll is used as a return value from [WalkDirFunc](https://pkg.go.dev/io/fs#WalkDirFunc) to indicate that all remaining files and directories are to be skipped. It is not returned as an error by any function.

https://cs.opensource.google/go/go/+/refs/tags/go1.24.3:src/path/filepath/path.go;l=429-431

Reply:
That seems like a huge anti-pattern in the Go standard library to use errors for control flow 😞